File: templates/Uwp/_composition/_shared/Project.TabbedPivot/Behaviors/PivotBehavior.cs
using System;
using System.Linq;
using Microsoft.Xaml.Interactivity;
using Windows.UI.Xaml.Controls;
using wts.ItemName.Helpers;

namespace wts.ItemName.Behaviors
{
    public class PivotBehavior : Behavior<Pivot>
    {
        protected override void OnAttached()
        {
            base.OnAttached();
            AssociatedObject.SelectionChanged += OnSelectionChanged;
        }

        protected override void OnDetaching()
        {
            base.OnDetaching();
            AssociatedObject.SelectionChanged -= OnSelectionChanged;
        }

        private async void OnSelectionChanged(object sender, SelectionChangedEventArgs e)
        {
            var removedItem = e.RemovedItems.Cast<PivotItem>()
                .Select(i => GetPivotPage(i)).FirstOrDefault();

            var addedItem = e.AddedItems.Cast<PivotItem>()
                .Select(i => GetPivotPage(i)).FirstOrDefault();

            if (removedItem != null)
            {
                await removedItem.OnPivotUnselectedAsync();
            }

            if (addedItem != null)
            {
                await addedItem?.OnPivotSelectedAsync();
            }
        }

        private static IPivotPage GetPivotPage(PivotItem pivotItem)
        {
            if (pivotItem.Content is Frame frame)
            {
                if (frame.Content is IPivotPage pivotPage)
                {
                    return pivotPage;
                }
            }

            return null;
        }
    }
}

File: templates/Uwp/_composition/_shared/Project.TabbedPivotVB/Behaviors/PivotBehavior.vb
Imports System
Imports Microsoft.Xaml.Interactivity
Imports wts.ItemName.Helpers

Namespace Behaviors
    Public Class PivotBehavior
        Inherits Behavior(Of Pivot)

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.SelectionChanged, AddressOf OnSelectionChanged
        End Sub

        Protected Overrides Sub OnDetaching()
            MyBase.OnDetaching()
            RemoveHandler AssociatedObject.SelectionChanged, AddressOf OnSelectionChanged
        End Sub

        Private Async Sub OnSelectionChanged(sender As Object, e As SelectionChangedEventArgs)
            Dim removedItem = e.RemovedItems.Cast(Of PivotItem).Select(Function(pi) GetPivotPage(pi)).FirstOrDefault()
            Dim addedItem = e.AddedItems.Cast(Of PivotItem).Select(Function(pi) GetPivotPage(pi)).FirstOrDefault()
            If removedItem IsNot Nothing Then
                Await removedItem.OnPivotUnselectedAsync()
            End If

            If addedItem IsNot Nothing Then
                Await addedItem.OnPivotSelectedAsync()
            End If
        End Sub

        Private Function GetPivotPage(pivotItem As PivotItem) As IPivotPage
            Dim frame = TryCast(pivotItem.Content, Frame)
            If frame IsNot Nothing Then
                Return TryCast(frame.Content, IPivotPage)
            Else
                Return Nothing
            End If
        End Function
    End Class
End Namespace